FHEM/00_SIGNALduino.pm

```diff
@@ -2856,23 +2858,102 @@ function SD_plistWindow(txt)
   $(div).html(txt);
   $("body").append(div);
   var oldPos = $("body").scrollTop();
+  var element = document.getElementById("SD_protoCaption");
+  var caption = element.innerHTML;
+  var btxtStable = "";
+  var btxtBlack = "";
+  if (caption.substr(0,1) != "d") {
+    btxtStable = "stable";
+  }
+  if (caption.substr(-1) == ".") {
+    btxtBlack = " except blacklist";
+  }
   $(div).dialog({
     dialogClass:"no-close", modal:true, width:"auto", closeOnEscape:true, 
     maxWidth:$(window).width()*0.9, maxHeight:$(window).height()*0.9,
-    buttons: [{text:"OK", click:function(){
-      $(this).dialog("close");
-      $(div).remove();
-      location.reload();
-    }}]
+    title: "Protocollist Overview",
+    buttons: [
+      {text:"select all " + btxtStable + btxtBlack, click:function(){
+           //FW_cmd(FW_root+ \'?XHR=1&cmd={SIGNALduino_FW_selectAll()}\', function(data){SD_selectAll(data)});
+      }},
+      {text:"deselect all", click:function(){
+           // FW_cmd(FW_root+ \'?XHR=1&cmd={SIGNALduino_FW_deselectAll()}\', function(){SD_deselectAll()});
+           SD_deselectAll()
+      }},
+      {text:"save to whitelist and close", click:function(){
+          var ret = SD_saveWhitelist();
+          FW_cmd(FW_root+ \'?XHR=1&cmd={SIGNALduino_FW_saveWhitelist("\'+ret+\'")}\');
+          $(this).dialog("close");
```

Comment:
Bei einem save würde ich jetzt nicht unbedingt das Fenster schließen

Reply:
Gibt es nach dem save noch Aktionen die der Anwender machen könnte?

Reply:
Vielleicht das Ergebnis ansehen.

Reply:
Das Ergebnis steht im Attribut whitelist, es sind die selben IDs, wie die wo mit der checkbox aktiviert wurden.

Reply:
Und wenn ich was aktiviere,  was aber dann doch nicht aktiviert wurde weil es ein anderes Attribut verhindert hat?
Die Liste soll doch anzeigen ob ein Protokoll aktiv ist und nicht ob es in der whitelistIDs enthalten ist.

Reply:
Du gehst von einer `sub SIGNALduino_IdList($@)` aus die noch nicht wie gewünscht funktioniert.
Mit den Anpassungen die ich an anderer Stelle schon vorgeschlagen habe, passt es.
Alles was in der whitelistIDs musst aktiviert werden.
Soll ich es hierher commiten oder dafür einen eigenen pullrequest machen.

Reply:
Die SIGNALduino_IdList funktioniert meiner Meinung nach schon so, dass dort ermittelt wird, welche Protokolle aktiv sind.